src/utils/hooks.ts

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { useCallback, useContext, useEffect, useMemo } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { OnboardingContext } from './OnboardingProvider';
+
+export const useGuardedOnboarding = () => {
+  const onboarding = useContext(OnboardingContext);
+  if (!onboarding)
+    throw new Error(
+      'Component should be wrapped inside the onboarding context',
+    );
+
+  const { flow, ...rest } = onboarding;
+  const { push } = useRouter();
+
+  useEffect(() => {
+    if (flow.length === 0) push('/onboarding/');
```

Comment:
I don't think that's necessary, since the server will handle the redirect, so it would effectively be the same as the onboarding home redirect. This implementation is cleaner imo as we have less `onboardingHome` imports thrown around, which means less to potentially refactor if we decide to change or remove the variable later on.